### ISSUE TITLE
On-board Peter Burkholder to cloud.gov

### ISSUE THREAD

**Issue by pburkholder** [18F/cg-product#1173]
# Onboarding Checklist

## Instructions

When someone new joins the cloud.gov team:

1. Create a new card in Trello in the cloud.gov program board assigned to the `Product Management` team called "Ramp up [NewPerson] on cloud.gov team".
2. View the raw source of this file.
3. Copy the text before the checklists to the Description field of the card.
3. Copy each applicable block of tasks into a its own new task list on the card, replacing "NewPerson" with the new person's name and  "Buddy" with the onboarding buddy's name.
4. Submit the card.
5. Assign the person who bravely volunteered to be the new person's Onboarding Buddy to the card.
6. Put the issue into the _Doing_ column.

---

In order to get NewPerson productively contributing to the cloud.gov team, Buddy should help NewPerson complete a prescribed set of tasks that will bring them up to speed.

## Directions

*NewPerson and Buddy:* Try to go through your checklists in order.

*Buddy:* If you can’t complete any of the items on your checklist personally, _you are responsible for ensuring that someone with the correct access completes that item_.

### Getting to know cloud.gov

* [ ] Take judicious notes on what about this onboarding process or cloud.gov is confusing or frustrating. If you notice a problem (especially with things like documentation), you are more than welcome to fix it! At the very least, please share this information with your buddy (or someone) at some point so we can make the team/platform better. (You can also file issues and pull requests on [the template for Onboarding issues](https://github.com/18F/cg-product/blob/master/OnboardingChecklist.md].)
* [ ] Figure out who your onboarding buddy is (they should reach out to you) and make sure this issue is assigned to them.
* [ ] Read [the team onboarding document](https://github.com/18F/cg-product/blob/master/Onboarding.md) for more context about cloud.gov.
* [ ] Bookmark the [pertinent links listed here](https://github.com/18F/cg-product/blob/master/PertinentLinks.md).
* [ ] Read the ["What is it?"](https://docs.google.com/presentation/d/1nCcti3dXG9TVGW3OqaWtnf96oXX8U8SBTM_WePFO_dg/edit#slide=id.p) presentation for a rundown of what cloud.gov is and does (specifically read slides 1-37; the rest are somewhat outdated)
* [ ] Read through [the Overview section of cloud.gov](https://cloud.gov/overview/) for a broader understanding of cloud.gov, especially as we present it to potential customers/users.
* [ ] Read the [January 2016 cloud.gov "Executive Business Case" document](https://docs.google.com/document/d/138OcG0Lt6gr9J0wM0TzzPNyTROmYAwfLIDujtweiwGw/edit#) for greater context about cloud.gov's potential impact in government.
* [ ] Read the [House Oversight and Government Reform Committee (July 12 2017) document](https://docs.google.com/document/d/1kDJdaPw7DSBPSa-XH-YsQpJVOmRKSK8sAghNhPFpegE/edit) for official answers to important questions.
* [ ] [Deploy a sample application to cloud.gov](https://cloud.gov/quickstart/) to get familiar with the basics of the PaaS from a user's perspective.
* [ ] We use Trello for [macro-level cadence and roadmapping](https://trello.com/b/03DgpOUQ/cloudgov-feature-kanban).
* [ ] Read the [Delivery Process document](https://github.com/18F/cg-product/blob/master/DeliveryProcess.md) to learn about how we work.
* [ ] Read our [service disruption guide](https://cloud.gov/docs/ops/service-disruption-guide/) to learn how we handle customer-facing service disruptions.
* [ ] Join [the cloud.gov team Google Group](https://groups.google.com/a/gsa.gov/forum/?hl=en#!forum/cloud-gov) so you can participate in team-wide internal communication.
* [ ] Join the [cloud.gov inquiries Google Group](https://groups.google.com/a/gsa.gov/forum/#!forum/cloud-gov-inquiries) so you can keep apprised of prospective new clients.
* [ ] Bookmark the [cloud.gov Google Drive folder](https://drive.google.com/drive/folders/0Bx6EvBXVDWwheUtVckVnOE1pRzA) -- that's where we put cloud.gov docs. If you create or move a doc there, it'll get the right access permissions for 18F team members to be able to view and edit it.
* [ ] Consider joining [Cloud Foundry's Slack channels](https://slack.cloudfoundry.org/) (we sometimes talk to folks in #gov and #gov-private –- you'll need a cloud.gov team member to invite you to the latter).
* [ ] If you haven't already, [request a license for Microsoft Office](https://handbook.18f.gov/office/) -- this is helpful for working with compliance documentation.
* [ ] Once you've finished the checklists below, make suggestions for steps that would have improved your onboarding experience as pull requests on [the onboarding checklist template](https://github.com/18F/cg-product/blob/master/OnboardingChecklist.md) used to make this issue.

### Required items for all team members

These items help us fulfill security and compliance requirements (including for FedRAMP). If you get stuck, or if these requirements are confusing, ask for help from your buddy or in a cloud.gov channel.

Pre-requisites:

* [ ] Complete [GSA OLU](https://gsaolu.gsa.gov/) GSA Mandatory Cyber Security and Privacy Training, including accepting the GSA IT Rules of Behavior, which is required before we can give you access to any cloud.gov systems. If you joined GSA more than two months ago, you've already completed this task and can just check the box.

Ensure you know what's happening on the team:

* [ ] Subscribe to [the cloud.gov team calendar](https://calendar.google.com/calendar/embed?src=gsa.gov_0samf7guodi7o2jhdp0ec99aks@group.calendar.google.com&amp;ctz=America/Los_Angeles) (click the + in the bottom right) so you know when assorted team meetings are happening in the various squads. Tip: When you plan Out of Office time, make a calendar event for that on the cloud.gov calendar so that your teammates know you'll be away.
* [ ] Subscribe (through the GitHub watch function) to the [cg-site](https://github.com/18F/cg-site) GitHub repository notifications.
* [ ] Ask Program Manager or Director for access to the cg-supportstream Slack channel.

Learn our policies and procedures:

* [ ] Coordinate with your onboarding buddy to go through [Incident Response Training](https://docs.google.com/presentation/d/1AZjQE8zBzMRWZIFUuJPkJLted1ykGtALrLPoPRx5Vls/edit#slide=id.p) within 60 days of joining the team (and annually after that). This will cover the following document, which you should also review before or after training:
* [ ] Read the [Incident Response Guide](https://cloud.gov/docs/ops/security-ir/).
* [ ] Coordinate with your onboarding buddy to go through [nonpublic information training](https://docs.google.com/presentation/d/1uB4MlGCu8ZYUxjKVZKwicQ95MvLxaT4Mh93y6w79GPw/edit#slide=id.p) within 60 days of joining the team (and annually after that). This will cover the following documents, which you should also review before or after training:
* [ ] Review the [18F open source policy guidance about protecting sensitive information](https://github.com/18F/open-source-policy/blob/master/practice.md#protecting-sensitive-information).
  * [ ] Read our [sharing secret keys](https://cloud.gov/docs/ops/secrets/#sharing-secret-keys) policy.
  * [ ] Review the [18F requirements for password management](https://handbook.18f.gov/password-requirements/).
* [ ] If you're Cloud Ops or Business Unit: Coordinate with your onboarding buddy to go through Contingency Planning training within 10 days (and annually after that). This will cover the following document, which you should also review before or after training:
* [ ] Read the [Contingency Plan](https://docs.cloud.gov/ops/contingency-plan/).

Read the following documents as well, which explain our practices in a formal written way. Reach out to your buddy or the team if anything is unexpected or confusing.
* [ ] Read the [Continuous Monitoring Strategy](https://cloud.gov/docs/ops/continuous-monitoring/), particularly the [cloud.gov team responsibilities](https://cloud.gov/docs/ops/continuous-monitoring/#cloud-gov-team).
* [ ] Read the [Configuration Management Plan](https://cloud.gov/docs/ops/configuration-management/).
* [ ] Read the [18F Security Policies and Procedures](https://github.com/18F/compliance-docs). These documents explain the high-level policies and procedures we must comply with while running cloud.gov, sorted into security control "families" They explain that we follow GSA IT security policy, and they provide a summary of the procedures in our System Security Plan.

## Theme-specific items

For explanations of our theme names, see [this glossary](https://github.com/18F/cg-product#sub-teamsthemes-of-work).

### Platform-specific items

* [ ] Review the System Security Plan (the latest version lives on [Google Drive](https://drive.google.com/drive/u/0/folders/0B6fPl5s12igNX3JwR2xFZVpmek0); look for "cloud.gov System Security Plan (SSP)" as a `.docx` file). Of particular note for onboarding: Section 9 (System Description) and Section 10 (System Environment)
* [ ] Bookmark [the Platform kanban board view](https://cm-jira.usa.gov/secure/RapidBoard.jspa?rapidView=1926&view=detail&selectedIssue=CG-107).
* [ ] Join the `#cg-platform` channel on Slack.
* Learn about [Cloud Foundry](https://www.cloudfoundry.org/)
  * [ ] Watch [Build Your Own Private Cloud Foundry](https://www.youtube.com/watch?v=v85r4Hy3jbs) to learn about running Cloud Foundry.
  * [ ] Set up Cloud Foundry locally using [cfdev](https://github.com/cloudfoundry-incubator/cfdev) and push an app to it.
  * [ ] Join [the Cloud Foundry Slack](http://slack.cloudfoundry.org/) in case you need to ask questions of upstream developers.
* Learn about [BOSH](https://bosh.io/):
  * [ ] watch [this video](https://www.youtube.com/watch?v=2jpN1mSPZ4Q)
  * [ ] Complete this [tutorial for BOSH](https://mariash.github.io/learn-bosh/)
  * [ ] For a bit of levity, read [this](http://events.linuxfoundation.org/sites/events/files/slides/seven-stages-of-bosh.pdf)
* [ ] Get to know how UAA is deployed/integrated.
* Get familiar with [Terraform](https://www.terraform.io/)
  * [ ] Go through the [getting started guide](https://www.terraform.io/intro/getting-started/install.html).
  * [ ] Examine [our provisioning procedures](https://github.com/18F/cg-provision)
* Learn about [Concourse](https://concourse.ci/)
  * [ ] Try a [tutorial](https://github.com/starkandwayne/concourse-tutorial).
  * [ ] Check out our [staging](https://ci.fr-stage.cloud.gov/) and [production](https://ci.fr.cloud.gov) Concourse instances, and take a look at some of [our pipelines](https://github.com/search?q=org%3A18F+cg-deploy).
  * [ ] Join [the Concourse Discord](https://discordapp.com/invite/MeRxXKW) in case you need to ask questions of upstream developers.
* Learn about [Kubernetes](https://kubernetes.io/)
  * [ ] Watch [The illustrated Children's Guide to Kubernetes](https://youtu.be/4ht22ReBjno)
  * [ ] Go through the [official tutorial](https://kubernetes.io/docs/tutorials/kubernetes-basics/)
  * [ ] Read the cloud.gov guide for [troubleshooting k8s](https://cloud.gov/docs/ops/runbook/troubleshooting-kubernetes/)
* [ ] Join [the cloud.gov operations notifications Google Group](https://groups.google.com/a/gsa.gov/forum/?hl=en#!forum/cloud-gov-notifications), so you can see alert information if OpsGenie is unavailable
* Get familiar with our AWS setup
  * [ ] Read about [AWS onboarding](https://cloud.gov/docs/ops/aws-onboarding/)
  * [ ] Read about the [AWS CLI](https://aws.amazon.com/cli/)
* [ ] Review the [daily maintenance tasks](https://cloud.gov/docs/ops/maintenance-list/) for when you're on support duty
* [ ] Make sure that OpsGenie will alert you through at least one of these ways:
  * [ ] Install/configure the app on the phone, go to the app settings, and change the alert for Low-Urgency alerts to actually alert you (make noise, vibrate, etc).
  * [ ] Make sure that OpsGenie is set up to alert you via email, and make sure that when it sends you an alert, it gets into your inbox, not spam.  If it gets into spam, releasing the spam seems to make it so that future emails will not make it there.

### Customer-specific items

* [ ] Join the `#cg-customer` channel on Slack.
* [ ] Ping @standup-bot for instructions on cg-customer standup.
* [ ] Review the [Customer board](https://cm-jira.usa.gov/secure/RapidBoard.jspa?rapidView=1929&view=detail&selectedIssue=CG-360).
* [ ] Bookmark link to [design folder](https://drive.google.com/drive/u/1/folders/0BwLqM4Nicmq-bUt0NjRjclFMUEU).
* [ ] Review the primary cloud.gov sites: [the dashboard](https://dashboard.cloud.gov/#/), [main landing page](https://cloud.gov/), and [documentation](https://cloud.gov/docs/).
* [ ] Read and bookmark our [Google Doc on cloud.gov Analytics](https://docs.google.com/document/d/1gSbP2ak2a3QLpCZIF_KlbQ2QHE6RjDI-7ZnnrJZvMDE/edit)
* [ ] [Request access to 18F Google Analytics](https://handbook.18f.gov/google-analytics/), so you can view cloud.gov site analytics ([including for the dashboard](https://docs.google.com/document/d/1gSbP2ak2a3QLpCZIF_KlbQ2QHE6RjDI-7ZnnrJZvMDE/edit)). Also request access to [Google Tag Manager](https://tagmanager.google.com/) which we mostly use to manage event tracking in Google Analytics.
* [ ] Ask the Program Manager for an invite to a DigitalGov Search account for cg-site, so you can configure it and view analytics for the most common searches.
* [ ] Ask the Program Manager for access to [GovDelivery](https://admin.govdelivery.com/administrators) so you can send announcements to our mailing lists

### If developing

* [ ] Review the [dashboard contributing guide](https://github.com/18F/cg-dashboard/blob/master/CONTRIBUTING.md) and [`cg-style` standards](https://github.com/18F/cg-style/blob/master/documentation/frontend_standards.md) which help frame what we're looking for in code review.
* [ ] Set up the [landing/docs page](https://github.com/18F/cg-site), [dashboard](https://github.com/18F/cg-dashboard), and [`cg-style`](https://github.com/18F/cg-style) locally.
* [ ] Set up `cg-style` to be [linked to the other sites locally](https://github.com/18F/cg-style#development-and-contributing-setup).
* [ ] Have a cloud.gov person send the `cg-dashboard` testing environment variables through Fugacious.

### For review

* [ ] Review the [design principles wiki](https://github.com/18F/cg-product/wiki) which explains the thinking behind the product and where that thinking comes from.
* [ ] Review the [design principles doc](https://docs.google.com/spreadsheets/d/14Y3RKaLUt6RPX5w13iz7oaSCpojEQ-Wqjnd8Ie_VkCc/edit#gid=259774738) to get a sense of how the cloud.gov team feels about the product.
* [ ] Review the [`cg-style` styleguide](https://pages.18f.gov/cg-style/) to get a sense of the global cloud.gov visual style.
* [ ] Review the dashboard: current [prod](https://dashboard.fr.cloud.gov/#/), [staging](https://dashboard-staging.app.cloud.gov/#/), and [demo](https://dashboard-demo.app.cloud.gov/#/).

### Business unit-specific items

* [ ] Bookmark the [BizOps board](https://trello.com/b/MSA7McuX).
* [ ] Join #products-platforms and all of the #cloud-gov-[everything] channels (it's ok to mute or leave some later).
* [ ] Ask Program Manager or Director for access (and ownership if appropriate) to the [cloud-gov-inquiries](https://groups.google.com/a/gsa.gov/forum/#!forum/cloud-gov-inquiries), [cloud-gov-support](https://groups.google.com/a/gsa.gov/forum/#!forum/cloud-gov-support), [cloud-gov-notifications](https://groups.google.com/a/gsa.gov/forum/#!forum/cloud-gov-notifications), and [cloud-gov-emergency](https://groups.google.com/a/gsa.gov/forum/#!forum/cloud-gov-emergency) groups.
* [ ] Read [how the cloud-gov-emergency group works](https://docs.google.com/document/d/1O5UW1M-XX8YIZJV1OcF1EqGWjPnNUi7L0mZpJZ6d-cs/edit#), and set up push notifications for these emails from your work smartphone if appropriate for your role.
* [ ] Ask #admins-salesforce for access to [Salesforce](https://gsa-peo.my.salesforce.com).
* [ ] In Salesforce, bookmark the [cloud.gov opportunities](https://gsa-peo.lightning.force.com/one/one.app#/sObject/00Ot0000000mFIxEAM/view?t=1484708635668) report.
* [ ] [If not also Cloud Ops] Ask Program Manager or Director for view-only access to [admin UI](https://admin.fr.cloud.gov).
* [ ] Ask #acquisition for the Acquisition NDA so you can sign it (if you haven't already).
* [ ] [If not also Cloud Ops] If appropriate for your role, ask for access to billing info in commercial AWS account from Program Manager or Director.

### New Director (System Owner) items

* [ ] Ask #tock to list you as the project contact for cloud.gov lines.
* [ ] Ask previous Director to give you Owner access to Nessus Manager (this requires the current primary contact access to Nessus, and then the owner of the license key also needs to change to the new director so that the new director can reset license keys - lastly, the new director needs to be made the primary contact).
* [ ] Ask previous Director to give you Owner access in Zendesk.
* [ ] Ask previous Director to give you [AWS account ownership](https://cloud.gov/docs/ops/aws-onboarding/) for GovCloud.
* [ ] Ask Program Manager or previous Director to give you cloud.gov calendar ownership.
* [ ] Ask Program Manager or previous Director to give you owner permission on the Google Groups. This should include [our Google Groups for our AWS accounts](https://docs.google.com/document/d/110o1L7EOby3hvE5d-cDhg2LBLHymbZLnMPe9kuk4qp8/edit#).
* [ ] Ask Program Manager or previous Director to let TTS Infrastructure know you are the new technical contact for cloud.gov.
* [ ] Ask Cloud Ops for read-write access to Admin UI.
* [ ] Review the [GSA Information Technology (IT) Security Policy](https://insite.gsa.gov/portal/content/520281) to understand our responsibilities as part of GSA.
* [ ] Review the [cloud.gov System Security Plan version 1.33](https://drive.google.com/drive/folders/0B6fPl5s12igNX3JwR2xFZVpmek0).

### Non team-specific items

_Note: These are items that do not fall into the boundary of work for a specific team. However, please consult your onboarding buddy to verify that your work relates to these items prior doing these two checklists._

### Compliance-specific items

* [ ] Join the `#cg-compliance` channel on Slack.
* [ ] Watch [Handling FISMA Faster and Better](https://www.youtube.com/watch?v=T1S52B1-NT4) for important context and background on the federal regulatory context in which cloud.gov operates.
* [ ] Do a cursory read of [Before You Ship](https://pages.18f.gov/before-you-ship/).

### Services-specific items

* [ ] Join the `#cg-services` channel on Slack.
* [ ] Read about Cloud Foundry [services from a user perspective](http://docs.cloudfoundry.org/devguide/services/).
* [ ] Read about [implementing services](http://docs.cloudfoundry.org/services/).
* [ ] Take a look at [our existing brokers](https://github.com/18F?utf8=%E2%9C%93&amp;query=broker).
* [ ] Learn about [Concourse](https://concourse.ci/)
  * [ ] Try a [tutorial|https://github.com/starkandwayne/concourse-tutorial].
  * [ ] Check out our [staging](https://ci.fr-stage.cloud.gov/) and [production](https://ci.fr.cloud.gov) Concourse instances, and take a look at some of [our pipelines](https://github.com/search?q=org%3A18F+cg-deploy).
* [ ] Join [the Cloud Foundry Slack](http://slack.cloudfoundry.org/).

## Buddy checklist

* [ ] Introduce yourself to the new team member and give them some of your background so they know who you are.
* [ ] Identify a straightforward, well-groomed story in progress that involves their skills domain, schedule a meeting with the owner for an introduction (if it's not you), and setup pairing sessions several times in the first week on the project.
* [ ] Identify a straightforward, well-groomed first story, ideally something they could conceivably complete in their first two/three weeks using their existing skills. Discuss the context with them, then make them the assignee for the card.
* [ ] Discuss suggestions for how the onboarding experience could have been improved and open as PRs on [the onboarding template](https://github.com/18F/cg-product/blob/master/OnboardingChecklist.md).
* [ ] Ask the Program Manager or Director to add the person to [Zendesk](https://cloud-gov.zendesk.com), so they can see how we handle non-18F support and read technical discussions happening with outside groups.
* [ ] Invite them to the private Slack channel cg-supportstream, used for backchanneling on support interactions.

## Required items for all team members

These items help us fulfill security and compliance requirements (including for FedRAMP).

* [ ] Make sure they're in [the list of people working on the project](https://docs.google.com/spreadsheets/d/1mW3tphZ98ExmMxLHPogSpTq8DzYr5Oh8_SHnOTvjRWM/edit#gid=0).
* [ ] Add their name, whether they're Cloud Ops (Platform), and the date they joined the team to the [training tracker](https://docs.google.com/spreadsheets/d/1hqU6cNeEB293OT0j3OvbdAFRkrf2zDOrPVxGfnr4sSw/edit#gid=0). Copy the formulas for the due dates from an existing row (grab the "corner" of the cells and pull down).
* [ ] As they complete training, fill out their completion dates in the [training tracker](https://docs.google.com/spreadsheets/d/1hqU6cNeEB293OT0j3OvbdAFRkrf2zDOrPVxGfnr4sSw/edit#gid=0).
* [ ] Add them to the @cloud-gov-team [in Slack’s Team Directory](https://get.slack.help/hc/en-us/articles/212906697-User-Groups#edit-a-user-group).
* [ ] Add them to the recurring cloud.gov meetings that are relevant for them in [the team calendar](https://calendar.google.com/calendar/embed?src=gsa.gov_0samf7guodi7o2jhdp0ec99aks@group.calendar.google.com&amp;ctz=America/Los_Angeles).
* [ ] Ask one of our cloud-gov team Github Maintainers (Britta or Eddie) to add them to the [@18F/cloud.gov team](https://github.com/orgs/18F/teams/cloud-gov) on GitHub.
* [ ] [Request a Jira account for NewPerson in the cloud.gov team.](https://cm-jira.usa.gov/projects/JAT/summary). You need to supply their name, email address, and phone number.
* [ ] Talk to the leadership about whether the #cg-team-priv history should be wiped, then invite the new person to that channel

## Platform-specific required items

* [ ] Help them review and understand the responsibilities of becoming a Cloud Operations team member, as listed in our SSP.
* [ ] Ask the Program Manager (Britta) or System Owner (Eddie) to have them added to the [@18F/cloud-gov-ops](https://github.com/orgs/18F/teams/cloud-gov-ops) team on GitHub. *(For contractors: Confirm they have cleared GSA security review before doing this one!)*
* [ ] If the new person is a contractor, ask the Program Manager (Britta) or System Owner (Eddie) to have them added to the [@18F/cloud-gov-contractors](https://github.com/orgs/18F/teams/cloud-gov-contractors) team on GitHub.
* [ ] Add them to @cg-operators [in Slack’s User Groups](https://get.slack.help/hc/en-us/articles/212906697-User-Groups#edit-a-user-group).
* Grant them access to the following:
  * [ ] [AWS Accounts](https://cloud.gov/docs/ops/aws-onboarding/) via the AWS web console (not Terraform) and provide one-time credentials.
  * [ ] AWS GovCloud
  * [ ] AWS East tied to GovCloud
  * [ ] AWS Admin Access
  * Note: AWS user names should be identical across accounts so that permissions can be correctly managed by terraform
  * [ ] Google Cloud Platform (GCP) Projects (by asking Derrick Rogers in GSA IT)
  * [ ] Nessus Manager GUI
  * [ ] Opsgenie
  * [ ] StatusPage
* [ ] [Make them an admin](https://cloud.gov/docs/ops/managing-users/#managing-admins) of the platform.
* [ ] Add them to the `cloud-gov-operators` organization in cloud.gov.
* [ ] Take them through [AWS onboarding](https://cloud.gov/docs/ops/aws-onboarding/).
* [ ] Add them to the set of [users in CloudCheckr](https://app.cloudcheckr.com/Users) and make them a member of [the cloud.gov group](https://app.cloudcheckr.com/Admin/UserGroupBuilder/fb111fab-ef5d-48d0-9472-cff691e1bd9c).
* [ ] Ask #infrastructure to add them to [the cloudgov subteam in Docker Hub](https://hub.docker.com/u/18fgsa/dashboard/teams/?team=cloudgov).
* [ ] Give them a walkthrough of cloud.gov from an architecture and repository perspective, focusing on SSP diagrams, external git repository and bosh.io dependencies, and our continuous delivery process with Concourse.
